lib/source_file.cpp

```diff
@@ -69,7 +71,35 @@ void InputSources::addComment(SourceInfo srcInfo, bool singleLine, cstring body)
     if (!singleLine)
         // Drop the "*/"
         body = body.exceptLast(2);
-    comments.push_back(new Comment(srcInfo, singleLine, body));
+    auto srcPos = SourcePosition(srcInfo.getStart());
+
+    comments.push_back(new Comment(srcPos, singleLine, body));
```

Comment:
construct `srcPos` in place (as a temporary) or move in place to take advantage of moving.